lib/Installation/Partitioner/LibstorageNG/v4/ExpertPartitionerController.pm

```diff
@@ -190,4 +209,26 @@ sub set_new_partition_size {
     $self->get_edit_partition_size_page()->press_next();
 }
 
+sub encrypt_partition {
+    my ($self, $args) = @_;
+    $self->get_expert_partitioner_page()->select_item_in_system_view_table($args->{partitioner_table});
+    send_key('down');    # on the current proposal the disks are not unfolded so we find the partition table and move one step down
```

Comment:
isn't it possible to create a function for pressing 'down', as well?

Reply:
i see that coming :) the answer is yes. Thing is i was planning to refactor a few things until i realize that it will mess with the PR and it was out of scope of the ticket. I decided to file a ticket for the refactoring. The reason is that the framework starts to  not inherits the SOLID principles. For instance, the `select_item_in_system_view_table` should have been responsible for get the right item from the table. In v4 the Hard-disks items are not collapsed as it is in v3 for instance. what i try to say is that it needs a more sophisticate approach than just wrap the send_key with a function, at the end. for example i would change the `expand_item_in_system_view_table` or i would create a `traverse_tree` or something to handle it. But i was afraid that it would touch other places too and it would require more work. thus the new ticket for the refactoring(as there are other places that they need some attention as well)

Reply:
So, this was done the easiest way possible because it's temporary?

Reply:
Yes. the most straight forward without break any other test for the moment.

Reply:
I don't get neither why here there is a hardcoded key, code should be uniform in that sense (even if there is some space for better design). The comment didn't help me to understand, because the partition table is some other specific flow, cannot just select the disk here?

Reply:
no it cant. it is folded and the code is not designed for this case. there is the `expand_item_in_system_view_table` but it is used once you have reach the item from `select_item_in_system_view_table`. to not break other tests for now this is just the most clear thing to do without unnecessary refactoring

Reply:
I also would propose doing it right, because it breaks MVC pattern and will lead to the mixing things all together, which is not hard with openQA. From what I read here, would not it work if this step is done in `select_item_in_system_view_table`? Not to break other tests, we can validate that list is not collapsed.